python/py_unzip/rezipper.py

```diff
@@ -0,0 +1,71 @@
+"""
+Repack a bazel python_zip_file (the same thing generated by specifying 'bazel build
+--build_python_zip') into a tarfile that extracts into a runnable application.
+"""
+
+import argparse
+import io
+import pathlib
+import tarfile
+import zipfile
+
+
+def main():
+    args = _parse_args()
+    run(src=args.src, dst=args.dst, package_dir=args.package_dir, main=args.main)
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument(
+        "--src",
+        help="Path to the source .zip file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--dst",
+        help="Path to the output .tar file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--package-dir",
+        help="The directory in which to expand the specified files.",
+        type=pathlib.Path,
+        required=True,
+    )
+    parser.add_argument(
+        "--main",
+        help="The __main__.py for the output tar.",
+        type=pathlib.Path,
+        required=True,
+    )
+
+    return parser.parse_args()
+
+
+def run(src: str, dst: str, package_dir: pathlib.Path, main: pathlib.Path):
+    with zipfile.ZipFile(src, "r") as z_in:
+        with tarfile.TarFile(dst, "w") as z_out:
+            addfile = _make_addfile(z_out, package_dir)
+
+            for info in z_in.infolist():
+                # Override the __main__.py from the python_zip_file.
+                if info.filename == "__main__.py":
+                    addfile(info.filename, main.read_bytes(), mode=0o755)
+                else:
+                    addfile(info.filename, z_in.read(info.filename), mode=0o644)
```

Comment:
can we not just copy the mode from the input file, or is that not available in zipfiles? I'd imagine there might be some cases where 644 isn't the only valid set of permissions

Reply:
when I first attempted it I found some oddities where bazel complained about `"permission denied"` but I should try again now that everything is working and I can triage one thing at a time